spec/support/fixtures/foo

```diff
@@ -477,5 +480,11 @@ Foo::CLI::Commands.register "generate webpack", Foo::Webpack::CLI::Generate
 Foo::CLI::Commands.register "hello",            Foo::Webpack::CLI::Hello
 Foo::CLI::Commands.register "sub command",      Foo::Webpack::CLI::SubCommand
 
+# we need to be sure that command will not override with nil command
+Foo::CLI::Commands.register "generate webpack", nil
+
+Foo::CLI::Commands.after("sub command") { puts 'After command hook' }
+Foo::CLI::Commands.before("sub command") { puts 'before command hook' }
```

Comment:
@jodosha finally I got something like this. WDYT about API? Now I want to cleanup all code and add specs